### PR TITLE
simplify `applyConcatenatedProperties`

### PR DESCRIPTION
**Pull request by bekzod**

### CODE REVIEW

**Comment by bekzod** [comment]
ping

File: packages/ember-metal/lib/mixin.js [hunk 1]
```diff
@@ -135,19 +135,7 @@ function giveMethodSuper(obj, key, method, values, descs) {
...
-  } else if (isArray(baseValue)) {
-    if (value === null || value === undefined) {
-      ret = baseValue;
-    } else {
-      ret = a_concat.call(baseValue, value);
-    }
-  } else {
-    ret = a_concat.call(makeArray(baseValue), value);
-  }
+  let ret = makeArray(baseValue).concat(makeArray(value));
```
Comment (bekzod) [inline packages/ember-metal/lib/mixin.js]: `makeArray` already has `null/undefined` checks which was done in `applyConcatenatedProperties` before

**Comment by stefanpenner** [comment]
cc @krisselden

File: packages/ember-metal/lib/mixin.js (hunk 1 above)
Comment (krisselden) [reply packages/ember-metal/lib/mixin.js]: there was a performance gain from avoiding the concat when it wasn't needed, this would need to be retested

**Comment by krisselden** [comment]
needs an https://github.com/krisselden/ember-macro-benchmark/

**Comment by bekzod** [comment]
benchmark: https://cl.ly/2e1k0Y3q2X3o/results.pdf

**Comment by mmun** [comment]
Benchmark looks good.

**Pull request merged**